crates/biome_js_analyze/src/lint/nursery/no_vue_duplicate_keys.rs

```diff
@@ -0,0 +1,193 @@
+    /// <script>
+    /// export default {
+    ///     computed: {
+    ///         count() {
+    ///             return this.value * 2;
+    ///         }
+    ///     },
+    ///     methods: {
+    ///         count() {
+    ///             this.value++;
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```vue
+    /// <script>
+    /// export default {
+    ///     props: ['foo'],
+    ///     data() {
+    ///         return {
+    ///             bar: 'baz'
+    ///         };
+    ///     },
+    ///     methods: {
+    ///         handleClick() {
+    ///             console.log('unique key');
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    /// ```vue
+    /// <script>
+    /// export default {
+    ///     computed: {
+    ///         displayMessage() {
+    ///             return this.message.toUpperCase();
+    ///         }
+    ///     },
+    ///     methods: {
+    ///         clearMessage() {
+    ///             this.message = '';
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    pub NoVueDuplicateKeys {
+        version: "next",
+        name: "noVueDuplicateKeys",
+        language: "js",
+        recommended: true,
+        severity: Severity::Error,
+        domains: &[RuleDomain::Vue],
+        sources: &[RuleSource::EslintVueJs("no-dupe-keys").same()],
+    }
+}
+
+impl Rule for NoVueDuplicateKeys {
+    type Query = VueComponentQuery;
+    type State = RuleState;
+    type Signals = Box<[Self::State]>;
+    type Options = NoVueDuplicateKeysOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let Some(component) = VueComponent::from_potential_component(
+            ctx.query(),
+            ctx.model(),
+            ctx.source_type(),
+            ctx.file_path(),
+        ) else {
+            return Box::new([]);
+        };
+
+        let mut key_declarations: FxHashMap<String, Vec<VueDeclaration>> = FxHashMap::default();
+
+        // Collect all declarations across all Vue component sections
+        for declaration in component.declarations(VueDeclarationCollectionFilter::all()) {
+            if let Some(name) = declaration.declaration_name() {
+                let key = name.text().to_string();
+                key_declarations.entry(key).or_default().push(declaration);
+            }
+        }
+
+        // Find duplicates
+        key_declarations
+            .into_iter()
+            .filter_map(|(key, declarations)| {
+                if declarations.len() > 1 {
+                    Some(RuleState { key, declarations })
+                } else {
+                    None
+                }
+            })
+            .collect::<Box<[_]>>()
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let first_declaration = state.declarations.first()?;
```

Comment:
Create the iterator, call next() instead of first(), and consume the iterator inside the loop. It's the whole point of iterators :)

Reply:
Good point, updated.